### PR TITLE
Apparmor 1.30

### PR DESCRIPTION
**Pull request by tallclair**
AppArmor is going to GA in v1.30, including some API changes.

**Note to reviewers:**
This PR is based on https://github.com/kubernetes/website/pull/45471, which should be merged first. Only the second commit should be reviewed in this PR.

https://github.com/kubernetes/enhancements/issues/24

/sig node
/milestone v1.30

### CODE REVIEW

**Comment by drewhagen** [comment]
Hello @tallclair 👋 please take a look at [Documenting for a release - PR Ready for Review](https://kubernetes.io/docs/contribute/new-content/new-features/#pr-ready-for-review) to get your PR ready for review before Tuesday March 12th 2024 18:00 PST. Thank you!

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:11]:
```suggestion
{{< feature-state feature_gate_name="AppArmor" >}}
```

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:57]: (nit) kubelet, lower case.

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:67]: (nit) kubelet, lower case.

**Review by sftim — COMMENTED** [review]
This is a good improvement, but: we should also mention AppArmor in either https://kubernetes.io/docs/concepts/workloads/pods/ or one of its sub pages.

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:276]: (nit)
```suggestion
### AppArmor profile within security context {#appArmorProfile}
```

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:285]:
```suggestion
for all containers in the pod (including init, sidecar, and ephemeral containers). If both a pod & container
```

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:283]:
```suggestion
You can specify the `appArmorProfile` on either a container's `securityContext` or on a Pod's
```
capital P if talking about the API kind

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:247]: Having regard to https://kubernetes.io/docs/contribute/style/style-guide/#avoid-statements-about-the-future
```suggestion
Kubernetes {{< skew currentVersion >}} does not provide any built-in mechanisms for loading AppArmor profiles onto
```

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:281]: This could go before the level three heading. Up to you.

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [inline content/en/docs/tutorials/security/apparmor.md:274]:
```suggestion
## Specifying AppArmor confinement
```
(this page is supposed to be a tutorial)

**Review by sftim — COMMENTED** [review]
I missed these from the [earlier review](https://github.com/kubernetes/website/pull/45176#pullrequestreview-1932942297).

**Review by drewhagen — COMMENTED** [review]
Hello @tallclair 👋, it looks like you've already engaged feedback in review of this PR. Great job! Keep this up. 🚀

@saschagrunert @mccormickt @tallclair @liggitt @pjbgf @SergeyKanzhelev @dchen1107
When one of y'all have a moment, please give a review for technical accuracy. A friendly reminder that this PR needs doc review complete by Doc Freeze on **_March 26th 18:00 PT_** to get this into the release.

Thank you again @sftim for another impressive and thoroughly detailed review.
Here are a few other suggestions I have:

**Review by saschagrunert — APPROVED** [review]
LGTM contentwise

File: content/en/docs/tutorials/security/apparmor.md
Comment (sftim) [reply content/en/docs/tutorials/security/apparmor.md:87]: Could you ever end up, on supported Kubernetes, running a Pod that you thought was confined but not? (this depends on how Kubernetes 1.27 handles unrecognized fields).

**Comment by sftim** [comment]
Once this is rebased down to 1 commit, I recommend getting technical reviews (SIG Node, SIG Security).

File: content/en/docs/tutorials/security/apparmor.md
Comment (tallclair) [reply content/en/docs/tutorials/security/apparmor.md:87]: Yes. Using only the field prior to apiserver v1.30 will result in the field being dropped, and the default AppArmor profile will be used (probably most common for this to over-constrain, but could under-constrain if a custom profile is used).

After approximately k8s v1.34 supplying only the annotation will no longer work.

Comment (tallclair) [reply content/en/docs/tutorials/security/apparmor.md:87]: I think caution makes sense here.

**Comment by tallclair** [comment]
Feedback addressed. Is @saschagrunert's LGTM sufficient for technical review, or do I need someone else?

**Comment by tallclair** [comment]
Updated pod security standards, and security checklist.

@sftim - AppArmor config is a fairly niche knob, and most users shouldn't touch it. I'm not sure it makes sense in the top-level pods doc, and we don't (yet?) have a dedicated security context doc that I could find.  The [security checklist](https://kubernetes.io/docs/concepts/security/security-checklist/#apparmor) reference seems like the most relevant bit.

**Comment by pjbgf** [comment]
LGTM from a content perspective.

**Comment by tallclair** [comment]
I scrubbed the docs for other AppArmor references, and updated a few other content pages. PTAL.

**Comment by sftim** [comment]
> AppArmor config is a fairly niche knob, and most users shouldn't touch it. I'm not sure it makes sense in the top-level pods doc, and we don't (yet?) have a dedicated security context doc that I could find. The [security checklist](https://kubernetes.io/docs/concepts/security/security-checklist/#apparmor) reference seems like the most relevant bit.

We don't really tell people reading the Pods concept that they have the option to do more around securing their workloads. We should. It can be a direct link to the pages (AppArmor, seccomp, etc) or an indirect via a link to a new page that describes what integrations Kubernetes offers.

Anyway, no mention at all is too little. 

A bigger revision to https://kubernetes.io/docs/concepts/workloads/pods/ would be great, but: it's too much to ask. The smaller ask is to add one sentence to https://kubernetes.io/docs/concepts/workloads/pods/ that mentions AppArmor. Maybe within https://kubernetes.io/docs/concepts/workloads/pods/#what-s-next?

**Comment by sftim** [comment]
Revising the Pod concept to link to https://k8s.io/docs/concepts/security/cloud-native-security/ is another way to ensure we link to AppArmor (feel free to pick this option).

We try to do some checks when things graduate to beta (or, if we missed it at beta, to stable). One of those checks is to make sure that if you only read the concepts section, you're likely to learn about the existence thing we're graduating. Cross-references like the one I'm advocating give readers the confidence that they won't need to brute-force their way through reading all the docs to know that they've covered the relevant details.

**Comment by tallclair** [comment]
@sftim I agree, but is it OK if I open a separate PR for it, and we don't block this PR on adding the pods context?

**Comment by sftim** [comment]
> I agree, but is it OK if I open a separate PR for it, and we don't block this PR on adding the pods context?

Sure. That works; happy to take your word on this.

**Comment by drewhagen** [comment]
Hello @tallclair :wave:! I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on Tuesday March 26th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as `lgtm` and `approve` labels applied, without any unaddressed comments or concerns from SIG Docs.

The status of this enhancement is marked as at risk for docs freeze.

I do notice though that we have a `lgtm`, so this is really close!
@kubernetes/sig-docs-en-owners @saschagrunert @pjbgf Does this look ready for approval?

Thank you!

**Review by pjbgf — COMMENTED** [review]
/approve

**Pull request merged**
